### PR TITLE
fix(openai): keep optional parameters under strict mode instead of dropping them

### PR DESCRIPTION
**Pull request by jkomyno**
This PR:

- builds on top of https://github.com/ComposioHQ/composio/pull/4209 by @AseemPrasad, keeping its recursive `toStrictJsonSchema()` and Python parity while changing the mechanism so strict mode stops deleting parameters
- keeps every optional parameter under `strict: true`: properties become required and optional ones accept `null` (the emulation OpenAI documents), instead of dropping 42% of parameters across the 930-tool corpus; `type` arrays stay as they are, so nullable objects stay nullable
- sends tools whose schema strict mode cannot express (objects with arbitrary keys, `allOf`, `prefixItems`, dangling `$ref`s, non-object roots) with `strict: false` and a warning naming the tool and path, instead of narrowing them to empty closed objects
- adds `omitNullToolArguments()`: the strict providers drop a `null` argument only where the tool's own schema rejects it, so nullable fields keep an explicit `null`
- keeps local `$ref`/`$defs` (recursion included) under strict mode instead of inlining them
- brings the Python `OpenAIResponsesProvider(strict=True)` to parity: emits `strict`, calls the base initializer, mirrors the rewrite and null omission
- makes Mastra and openai-agents use the same strict semantics (`OpenAIAgentsProvider({ strict: true })` previously had no effect)
- pins the behavior with a shared `strict-cases.json` corpus (byte-identical TypeScript/Python copies) plus edge-case regression tests enumerated independently with a second model

## Context

OpenAI's structured-outputs contract accepts `"type": ["string", "null"]` and rejects `type` next to `anyOf`. Validated against OpenAI's own `toStrictJsonSchema` converter over the 930 real tool schemas in `ts/packages/cli/test/__mocks__/tools.json`: the previous approach was accepted for 930/930 tools but only after removing 1,682 properties; this one emits strict schemas for 816 tools (all accepted, no property lost, idempotent) and downgrades the 114 tools that use free-form or map-style parameters.

https://claude.ai/code/session_01TDrxCHn2hg51HmxVstSUgs

### CODE REVIEW

File: python/composio/core/provider/_openai_responses.py [hunk 1]
```diff
@@ -10,27 +10,97 @@
...
         }
+        if not self.strict:
+            return wrapped
+
+        # Structured outputs enforce their contract at every depth: all
+        # properties required, closed objects, no annotation keywords. The
+        # strict rewrite keeps every parameter (optional ones become nullable)
+        # and reports constructs it cannot express; such a tool is sent
+        # without strict mode rather than with a narrower schema.
+        source = parameters if tool.input_parameters else dict(_EMPTY_OBJECT_SCHEMA)
```
[the remark this answers is not in the document]
Comment (awaisalidccs7-ui) [reply python/composio/core/provider/_openai_responses.py]: Fixed in the current-`next` rescue PR #4266 at head `57fe9c98ac48947d7e82b24f69a41d87bf300bc5`. Python now distinguishes explicit `{}` from missing (`None`) input parameters, so `{}` remains a free-form schema and falls back to `strict: false`, matching TypeScript. A dedicated regression test is included. I have left this original thread unresolved because the original #4257 head itself has not moved.
